pac4j-http/src/main/java/org/pac4j/http/client/indirect/FormClient.java

```diff
@@ -124,7 +124,7 @@ private RequiresHttpAction handleInvalidCredentials(final WebContext context, fi
 		// it's an AJAX request -> unauthorized (instead of a redirection)
 		if (getAjaxRequestResolver().isAjax(context)) {
 			logger.info("AJAX request detected -> returning " + errorCode);
-			return RequiresHttpAction.unauthorized("AJAX request -> " + errorCode, context, null);
+			return RequiresHttpAction.configurable("AJAX request -> " + errorCode, errorCode, context);
```

Comment:
And I think the `handleInvalidCredentials` should no longer have an `errorCode` parameter.